### DISCUSSION TITLE
Pane padding?

### DISCUSSION THREAD

**Discussion by permelin** [wezterm/wezterm#1383] [Q&A] [answered]
I'm working on setting WezTerm up exactly how I want it, but I can't find a way to set padding/margin for panes? Is there none? I'm rather particular about margins, and having absolutely zero space between text and dividers is making me uncomfortable.

**Comment by wez** [comment, marked as answer]
Today the padding is at the window level, with pane dividers having 0.5 cell padding; you shouldn't be in a situation where the divider is right up against the text.

What sort of padding would you like to be able to specify?

**Reply by permelin** [reply]
Half a cell isn't nearly enough for my taste. "Absolutely zero space" was clumsy hyperbole.

Personally I only want to set padding/margins in pixels or points. I'm coming from Kitty and there you can set top/right/bottom/left separately both on window and panes. I have them at zero on the window, 12 pts left/right and 6 pts top/bottom on the panes. But Kitty has a configuration called `placement_strategy` that you can set to `center` which will vertically center each pane when the window size is not an exact cell multiple, so I get some extra pixels top/bottom from that. In WezTerm you only pad the window right/bottom for that.

It's a tricky question how to do this without having it explode into lots of knobs and sliders. For example, if I were one of those people who live in tmux, I would probably want the bottom of the pane flush with the divider/window (which you can get in Kitty with `placement_strategy bottom`).

**Reply by wez** [reply]
OK.  Relevant issues for you to track:

https://github.com/wez/wezterm/issues/195
https://github.com/wez/wezterm/issues/1124
https://github.com/wez/wezterm/issues/1176
https://github.com/wez/wezterm/issues/1370

**Comment by siliconwitch** [comment]
Would also love this feature. Right now I use window spacing to add a nice margin all the way around the terminal, but it doesn't seem possible to match the same margin on split panes. Out of curiosity, what would it take to implement this? I assume that any pane padding would need to be in whole cells rather than some arbitrary number of pixels?

<img width="1440" alt="Screenshot 2024-08-24 at 09 40 50" src="https://github.com/user-attachments/assets/22358950-71d6-42dd-9b2c-0ecb7fcd0d56">

**Comment by denppa** [comment]
While per pane padding, or gaps, is a nice to have, I am also looking for the ability to thicken the pane divider / separators. As they are very thin at the moment and difficult to see on certain screens.

**Comment by RodogInfinite** [comment]
I'd like to see something like window_frame: 

I created this function to change the color of the window border under specific conditions, like being in tab mode, which helps to avoid unintended behavior from forgetting what mode you're in:

```lua
function CreateWindowBorderToggle(color)
	local is_active = false

	return function(window)
		is_active = not is_active
		window:set_config_overrides({
			window_frame = {
				border_left_width = is_active and "0.25%" or "0px",
				border_right_width = is_active and "0.25%" or "0px",
				border_top_height = is_active and "0.25%" or "0px",
				border_bottom_height = is_active and "0.25%" or "0px",
				border_left_color = is_active and color or "transparent",
				border_right_color = is_active and color or "transparent",
				border_top_color = is_active and color or "transparent",
				border_bottom_color = is_active and color or "transparent",
			},
		})
		return is_active
	end
end
```

it would be nice to do the same for panes in pane mode something like:

```lua
function CreatePaneBorderToggle(color)
	local is_active = false

	return function(pane)
		is_active = not is_active
		pane:set_config_overrides({
			pane_frame = {
				border_left_width = is_active and "0.25%" or "0px",
				border_right_width = is_active and "0.25%" or "0px",
				border_top_height = is_active and "0.25%" or "0px",
				border_bottom_height = is_active and "0.25%" or "0px",
				border_left_color = is_active and color or "transparent",
				border_right_color = is_active and color or "transparent",
				border_top_color = is_active and color or "transparent",
				border_bottom_color = is_active and color or "transparent",
			},
		})
		return is_active
	end
end
```
